### PR TITLE
[tune] Simplify API

### PR DESCRIPTION
**Pull request by richardliaw**
Uses `tune.run` to execute experiments as preferred API.

@noahgolmant

This does not break backwards compat, but will slowly internalize `Experiment`. 

In a separate PR, Tune schedulers should only support 1 running experiment at a time.

### CODE REVIEW

File: doc/source/tune-schedulers.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@ By default, Tune schedules trials in serial order with the ``FIFOScheduler`` cla
 
 .. code-block:: python
 
-    tune.run_experiments({...}, scheduler=AsyncHyperBandScheduler())
+    tune.run( ... , scheduler=AsyncHyperBandScheduler())
```
Comment (hartikainen) [inline doc/source/tune-schedulers.rst]:
```suggestion
    tune.run(... , scheduler=AsyncHyperBandScheduler())
```

File: python/ray/tune/examples/async_hyperband_example.py [hunk 2]
```diff
@@ -63,24 +63,21 @@ def _restore(self, checkpoint_path):
...
-                "config": {
-                    "width": sample_from(
-                        lambda spec: 10 + int(90 * random.random())),
-                    "height": sample_from(
-                        lambda spec: int(100 * random.random())),
-                },
-            }
+    run(MyTrainableClass,
+        name="asynchyperband_test",
+        **{
```
Comment (hartikainen) [inline python/ray/tune/examples/async_hyperband_example.py]: Is this syntax python 2 compatible?

File: python/ray/tune/examples/mnist_pytorch.py [hunk 3]
```diff
@@ -165,27 +165,24 @@ def test():
...
-                "run": "train_mnist",
-                "num_samples": 1 if args.smoke_test else 10,
-                "config": {
-                    "lr": tune.sample_from(
-                        lambda spec: np.random.uniform(0.001, 0.1)),
-                    "momentum": tune.sample_from(
-                        lambda spec: np.random.uniform(0.1, 0.9)),
-                }
+    tune.run(
+        lambda cfg, rprtr: train_mnist(args, cfg, rprtr),
```
Comment (hartikainen) [inline python/ray/tune/examples/mnist_pytorch.py]:
```suggestion
        lambda config, reporter: train_mnist(args, config, reporter),
```

File: python/ray/tune/tune.py [hunk 4]
```diff
@@ -197,3 +238,58 @@ def run_experiments(experiments,
...
+
+    """
+    # This is important to do this here
+    # because it schematize the experiments
+    # and it conducts the implicit registration.
+    experiments = convert_to_experiment_list(experiments)
+
+    trials = []
+    for exp in experiments:
+        trials += run(
```
Comment (hartikainen) [inline python/ray/tune/tune.py]: Just to make sure: is this expected to return flattened list of trials or an array of trials per experiment?

File: python/ray/tune/tune.py [hunk 5]
```diff
@@ -35,39 +35,113 @@ def _make_scheduler(args):
             args.scheduler, _SCHEDULERS.keys()))
 
 
-def _find_checkpoint_dir(exp_list):
-    assert exp_list, "Experiments must be specified via `run_experiments`"
-    exp = exp_list[0]
-    # TODO(rliaw): Make sure this is resolved earlier.
+def _find_checkpoint_dir(exp):
+    # TODO(rliaw): Make sure the checkpoint_dir is resolved earlier.
```
Comment (hartikainen) [inline python/ray/tune/tune.py]: Is this a todo that should be resolved before merging this or just a note for future?

File: python/ray/tune/examples/async_hyperband_example.py (hunk 2 above)
Comment (ericl) [reply python/ray/tune/examples/async_hyperband_example.py]: I think so.

**Review by ericl — APPROVED** [review]

File: python/ray/tune/tune.py (hunk 5 above)
Comment (richardliaw) [reply python/ray/tune/tune.py]: Note for future (I just made it more concrete)

File: python/ray/tune/tune.py (hunk 4 above)
Comment (richardliaw) [reply python/ray/tune/tune.py]: I just tried to maintain the same behavior as before, which was to return a flattened list of trials

File: python/ray/tune/examples/async_hyperband_example.py (hunk 2 above)
Comment (richardliaw) [reply python/ray/tune/examples/async_hyperband_example.py]: (unfortunately not 👎, but fixed now)

**Review by hartikainen — APPROVED** [review]

**Pull request merged**